### PR TITLE
Fix authentication errors on AWS ECR

### PR DESCRIPTION
**Pull request by Rowern**
On AWS ECR we do need to use the Basic Authentication mechanism
because otherwise the scheme will be "bearer" (and not "basic)
and the check of vendor/github.com/docker/distribution/registry/client/auth/session.go#L98
will fail and no authentication header will be put in the request.

This will fix (#204)

### CODE REVIEW

**Comment by Rowern** [comment]
We might get away with just updating the `./assets/cacerts.pem` with the one from alpine `/etc/ssl/certs/ca-certificates.crt` from the latest `ca-certificates` package. What do you think ?

**Comment by evelynl94** [comment]
> We might get away with just updating the `./assets/cacerts.pem` with the one from alpine `/etc/ssl/certs/ca-certificates.crt` from the latest `ca-certificates` package. What do you think ?

Thats fine by me. `cacerts.pem` was generated from the ca-certificates package in jessie/stretch base image, maybe it is indeed outdated. Have you tried setting `SSL_CERT_DIR=/makisu-internal/certs/`?

**Comment by Rowern** [comment]
I got everything to work on my setup, here is how:
 - First solution, I thought of adding the `ca-certificates` package inside the built image (it did work, but it's really a nasty solution, it might work because the ECR helper binary is run in the context of the built image and not the host)
 - Second solution, update the ca-certificates of the makisu alpine image, but until I changed the env var `SSL_CERT_DIR=/makisu-internal/certs/` it did not work (thanks @evelynl94 for your help)

Another thing for people like us that need to assume a role (so that need a specific aws config file), you can put your config file inside the `/makisu-internal` dir and export the env var `AWS_CONFIG_FILE`.

To sum up here are the env var I needed to export in order to make my setup work (most people won't need the AWS_* env var):
`export AWS_SDK_LOAD_CONFIG=true AWS_PROFILE=prod AWS_CONFIG_FILE=/makisu-internal/.aws/config SSL_CERT_DIR=/makisu-internal/certs/` (after doing a `cp /etc/ssl/certs/ca-certificates.crt /makisu-internal/certs/cacerts.pem`)

**Comment by Rowern** [comment]
I just re-checked on a known failing Dockerfile (where the ca certificates are not valid):
```Dockerfile
FROM node:10-alpine

ADD Dockerfile .
```

Setting the `SSL_CERT_DIR=/makisu-internal/certs/` does work so the cacerts.pem is up to date, I'm going to remove it and update the documentation to explicitly say how to set this env var for ECR registry.

**Comment by evelynl94** [comment]
@Rowern thats awesome. Thanks for the fix and the README update too :)

As for `SSL_CERT_DIR`, originally I thought it was only used by the gcr credhelper, but I was wrong. It is a common env for ssl. We will add it to the makisu Dockerfiles.
